### PR TITLE
chore(deps): Update docusaurus from beta-16 to beta-17, including rewriting dark mode suppression

### PR DESCRIPTION
**Pull request by pepopowitz**
Partially addresses #1024 

This PR updates docusaurus dependencies from beta-16 to beta-17. As a necessary side-quest due to feature deprecation, it rewrites our dark-mode suppression according to [docusaurus's recommendation](https://github.com/facebook/docusaurus/pull/6771). 

## Proof that there's still no option to switch color modes

image

### What it would look like if the color mode switch wasn't suppressed

In other words, what it looks like if I delete the swizzled ColorModeToggle from this PR, to show that the file is actively doing something: 

https://user-images.githubusercontent.com/1627089/177629810-a64172f0-4911-43a0-9806-eb702ce332b2.mov

### CODE REVIEW

**Review by akeller — APPROVED** [review]
👍 

If you are interested in tackling the beast that is CSS & color theming, you can ask marketing (mainly Andy Buck) about recommendations to make dark mode with our docs a thing and add it to our backlog. I'm not anti-dark mode, I just don't think it works with our branding as-is.
